requires.py

```diff
@@ -48,6 +48,9 @@ def manage_flags(self):
         toggle_flag(
             self.expand_name('{endpoint_name}.default_cni.available'),
             self.is_joined and self.get_default_cni() is not None)
+        toggle_flag(
+            self.expand_name('{endpoint_name}.api_endpoints.available'),
+            self.is_joined and all(self.get_api_endpoints()))
```

Comment:
Why `all` here? This will set `{endpoint_name}.api_endpoints.available` even if there are no API endpoints, right?

```
>>> all([])
True
```

Reply:
Ah, good catch.  I'm not sure why I thought I needed `all()` there.